### DISCUSSION TITLE
zigbee2mqtt si riavvia continuamente

### DISCUSSION THREAD

**Discussion by nassuper** [indomus/forum#3200] [ZigBee, Z-Wave, Thread, Matter]
ciao,

sto provando a configurare un sistema per gestire la domotica di casa. Ho un minipc con openmediavault dove uso docker compose per installare quello che serve.
ho spostato la cartella data e compose su un secondo disco.
Ho già installato homeassistant, hacs e mosquitto e sembra che funzioni tutto correttamente, ma ho grossi problemi con zigbee2mqtt.
Vi allego il file compose utilizzato:

```
services:
    zigbee2mqtt:
        container_name: zigbee2mqtt
        image: koenkk/zigbee2mqtt
        restart: unless-stopped
        volumes:
            - CHANGE_TO_COMPOSE_DATA_PATH/zigbee2mqtt/data:/app/data
            - /run/udev:/run/udev:ro
        ports:
            - 8080:8080
        environment:
            - PUID=1001
            - PGID=100
            - TZ=Europe/Rome
        devices:
            - /dev/ttyUSB0:/dev/ttyUSB0
        network_mode: host
```

questo è il file di configurazione:

```
homeassistant: false
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://192.168.1.100'
  user: myuser
  password: mypassword
  include_device_information: true
serial: /dev/ttyUSB0
advanced:
  # verificare sia corretto anche se ho spostato i folders di compose su secondo disco
  log_directory: /var/log
experimental: {}
frontend:
  port: 8080
ota:
  update_check_interval: 10
disable_automatic_update_check: false
```

riuscite ad indirizzarmi sulla strada corretta per far funzionare anche zigbee2mqtt?  grazie ;)

**Comment by KC-inDomus** [comment]
`CHANGE_TO_COMPOSE_DATA_PATH/zigbee2mqtt/data`

non ti dice nulla? ;)

**Comment by nassuper** [comment]
si, è la stringa che permette a docker di andare a salvare /leggere i file nel percorso giusto. va praticamente a leggere il percorso dal corrispettivo campo all'interno di compose settings. quella stringa viene poi automaticamente sostituida dal programma con quella riportata nei settings.  fammi sapere se sbaglio :)

**Reply by KC-inDomus** [reply]
grazie di avermelo spiegato, lo so bene :) se hai usato `CHANGE_TO_COMPOSE_DATA_PATH/zigbee2mqtt/data` è normale che tu abbia problemi

**Comment by nassuper** [comment]
scusa, non te l'ho spiegato, ho detto come ho interpretato io questa cosa. non credo di doverlo sostituire con il percorso del folder condiviso. le altre cose installate con lo stesso sistema funzionano. ma essendo all'inizio posso benissimo aver fatto confusione. se mi spieghi un po' meglio forse capisco.  grazie ;)

**Reply by nassuper** [reply]
ho anche provati il percorso completo, ma non cambia nulla.

**Reply by snakuzzo** [reply]
>  ho detto come ho interpretato io questa cosa. non credo di doverlo sostituire con il percorso del folder condiviso. le altre cose installate con lo stesso sistema funzionano

Questa cosa è assolutamente impossibile quel `CHANGE_TO_COMPOSE_DATA_PATH` non esiste in nessun universo conosciuto. Il bind dei volumi funziona passando un path reale (relativo, assoluto o tramite variabile...questo dipende da te). Quindi questa cosa va cambiata.

Quanto al darti una mano, se non condividi qui (con markdown) i log di zigbee2mqtt è impossibile. Nessuno ha la palla di vetro per capire al volo il motivo del riavvio (al netto di quanto detto prima sul bind dei volumi, ovviamente).

Quindi è inutile che continui a chiedere una mano senza condividere i log

**Reply by snakuzzo** [reply]
e non creare nuovi thread nella stessa discussione ad ogni risposta. Fai reply sotto l'ultimo messaggio

**Reply by nassuper** [reply]
ok. allora riprovo a reinstallare tutto e a sostituire CHANGE_TO_COMPOSE_DATA_PATH con il vero percorso della cartella condivisa. sia per zigbee2mqtt che per gli altri containers.
poi vedo di trovare ilog di zigbee2mqtt e di capire come condividerli con markdown

**grazie** :)

**Reply by snakuzzo** [reply]
1. non serve reinstallare tutto
2. quanto a CHANGE_TO_COMPOSE_DATA_PATH devi sapere tu che directory hai e come hai organizzato le cose

**Reply by nassuper** [reply]
ho usato il percorso come segue:
volumes:
            - /srv/dev-disk-by-uuid-cb71ad38-2a22-422a-9194-96c7dfb1202m/data/docker/appdata/zigbee2mqtt:/app/data
            - /run/udev:/run/udev:ro
ma ho il solito problema.
che file devo allegare? quello che trovo in diagnostics -> system logs -> logs -> **syslog** ?

Grazie :)

**Reply by snakuzzo** [reply]
ma cos..!!!
Perdona la franchezza, ma credo tu non abbia capito assolutamente nulla di quello che stai facendo...

`CHANGE_TO_COMPOSE_DATA_PATH` è la directory locale (dell'host) che sarà in bind con la directory `/app/data` del container.

Es.

```
        volumes:
            - /home/mioutente/zigbee2mqtt/data:/app/data
            - /run/udev:/run/udev:ro
```

**Reply by snakuzzo** [reply]
Continui ad andare a caso...non sai cosa stai facendo.
Non devi postare il log del sistema operativo, ma quello del container.

Una volta adeguato come si deve, ricrea il container e se hai problemi posta il log del container (non devi allegare alcun file)
Per tua info, il log lo recuperi con...

```
docker logs <nome o id del container>
```

Domanda...sai cos'è docker ? Sai cos'è un container ? Perché a me sembra proprio di no

**Reply by snakuzzo** [reply]
In aggiunta, ti consiglierei di dare un'occhiata alle nostre guide...

https://indomus.it/formazione/gestire-applicazioni-docker-tramite-docker-compose/
https://indomus.it/guide/installare-e-configurare-zigbee2mqtt-con-docker-su-linux-debian/

**Reply by nassuper** [reply]
asp non sono molto pratico, ma mi impegno :)

allora io ho installato openmediavault su un mini pc. ho installato docker e compose. ho aggiunto un disco esterno sul quale ho spostato le cartelle di compose/docker. a questopunto ho iniziato ad installare home assistant, mosquitto e quindi zigbee2mqtt.
il percorso che ho riportato sopra è quello che vedo nella cartella condivisa:
disco esterno, data, docker, appdata e compose.
se io uso il terminal, posso navigare fino alle cartelle in questione.

dove sto sbagliando?

**Reply by nassuper** [reply]
le guide le ho lette e seguite, ma non prevedono l'uso di un secondo disco per docker compose. probabilmente è li che ho fatto casini

**Reply by snakuzzo** [reply]
Diciamo che non è solito usare "quel" percorso. Quando hai un FS esterno, solitamente lo monti su una directory che decidi tu (a seconda della distro, se previsto, viene montato in automatico su delle directory di default). Ad ogni modo...se sei convinto sia corretto e continui ad avere problemi tocca vedere il log di zigbee2mqtt come ti ho indicato prima

**Reply by nassuper** [reply]
ho seguito dei tutorial. ma alla fine dopo aver installato openmediavault come SO, ho solo formattato il disco, creato il file system e una cartella condivisa. il percorso me lo sono trovato così (scomodo e lungo). cerco di capire dove trovare il log di zigbee2mqtt. grazie

**Reply by snakuzzo** [reply]
cosa devi capire ?
Ti ho scritto già come fare!!!

```
docker logs <nome o id del container>
```

**Reply by snakuzzo** [reply]
e ti ho pure linkato la guida per linux prima...è inutile che vai a riguardarti quella per rpi

https://indomus.it/guide/installare-e-configurare-zigbee2mqtt-con-docker-su-linux-debian/

**Reply by nassuper** [reply]
grazie ;) 
mi compare questo:

```
If you don't know how to solve this, read https://www.zigbee2mqtt.io/guide/configuration

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!


Using '/app/data' as data directory
Starting Zigbee2MQTT without watchdog.


!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
            READ THIS CAREFULLY

Refusing to start because configuration is not valid, found the following errors:
- serial must be object

If you don't know how to solve this, read https://www.zigbee2mqtt.io/guide/configuration

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!


Using '/app/data' as data directory
Starting Zigbee2MQTT without watchdog.


!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
            READ THIS CAREFULLY

Refusing to start because configuration is not valid, found the following errors:
- serial must be object

If you don't know how to solve this, read https://www.zigbee2mqtt.io/guide/configuration

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!


Using '/app/data' as data directory
Starting Zigbee2MQTT without watchdog.


!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
            READ THIS CAREFULLY

Refusing to start because configuration is not valid, found the following errors:
- serial must be object

If you don't know how to solve this, read https://www.zigbee2mqtt.io/guide/configuration

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

```

**Reply by nassuper** [reply]
sembra che il problema stia ancora nel percorso, o sbaglio?

**Reply by snakuzzo** [reply]
Sbagli...perché c'è scritto chiaramente qual è il problema...

```
- serial must be an object
```

Se tu avessi letto le guide con attenzione, avresti notato questo:

![image](https://github.com/user-attachments/assets/d9f910d4-1b25-4a9a-b221-5b7e943f4ddc)

**Reply by nassuper** [reply]
ho anche installato lo script per trovare la porta dove ho inserito zigbee

```
/dev/bus/usb/003/004 - ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_18799a5f38b6ed22bc3cde0ea8339f5d
/dev/ttyUSB0 - ITead_Sonoff_Zigbee_3.0_USB_Dongle_Plus_18799a5f38b6ed22bc3cde0ea8339f5d
```

**Reply by snakuzzo** [reply]
parliamo e non ci capiamo!!!!
Hai sbagliato a scrivere la configurazione. E' proprio sintatticamente sbagliata!!!!!

**Reply by snakuzzo** [reply]
Famo così...altrimenti qui non ne usciamo più, e abbiamo già speso troppo tempo per sta "ca*ata"...
Tu hai scritto...

```yaml
serial: /dev/ttyUSB0
```

...ma dovevi scrivere (se hai la ZBDongle-P) ...

```yaml
serial:
  port: /dev/ttyUSB0
```

...ma dovevi scrivere (se hai la ZBDongle-E) ...

```yaml
serial:
  port: /dev/ttyUSB0
  adapter: ember 
```

**Reply by snakuzzo** [reply]
LEGGI COSA HO APPENA SCRITTO SOPRA!!!!

**Reply by nassuper** [reply]
azz!

vado subito a provare. grazie ;)

**Reply by nassuper** [reply]
porca miseria ....funziona!!! O almeno ora dice up.

Grazie infinite per la pazienza e l'aiuto. ci metto l'impegno, ma linux è ancora complicato per me.  c'è un modo per capire se homeassistant e mosquitto sono stati "installati" correttamente? apparentemente sembra andare tutto.

Grazie ancora per  tutto. ciao :)

**Reply by snakuzzo** [reply]
se funziona...funziona...poi ci sono i log.
E comunque linux non c'entra assolutamente nulla con il problema che hai avuto

**Reply by nassuper** [reply]
spero con un po' di pazienza e buona volontà di capire meglio come muovermi. appena finisco di mettere anche red, vado a fare gli esercizi.

grazie ancora e buona giornata :)
